### PR TITLE
[Shipa 1474] Pool -> Framework

### PR DESCRIPTION
**Pull request by stinkyfingers**
# Description

**NOTE** This PR builds on #87  - it should be merged first.

Addresses # [1474](https://shipaio.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=SHIPA&modal=detail&selectedIssue=SHIPA-1474)

Converts `Pool` in command/CRD/Controller/docs to `Framework`. Adds "pool" alias to "framework" command w/ a usage note.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (documentation addition or typo, file relocation)

## Testing

- [ ] New tests were added with this PR that prove my fix is effective or that my feature works (describe below this bullet)
- [x] This change requires no testing (i.e. documentation update)

## Documentation

- [x] All added public packages, funcs, and types have been documented with doc comments
- [ ] I have commented my code, particularly in hard-to-understand areas

## Final Checklist:

- [x] I followed standard [GitHub flow](https://guides.github.com/introduction/flow/) guidelines
- [x] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

## Additional Information (omit if empty)

Please include anything else relevant to this PR that may be useful to know.

### CODE REVIEW

File: ci/ketch-ci.sh [hunk 1]
```diff
@@ -131,12 +131,12 @@ ensure_resource 'ketch-controller-manager' 1
...
-    if [ -z "$POOL" ]; then usage "Pool name required"; fi;
+    if [ -z "$POOL" ]; then usage "Framework name required"; fi;
     if [ -z "$INGRESS_ENDPOINT" ]; then usage "Ingress endpoint required"; fi;
 
-    # Add a pool with ingress Traefik (default), replace ingress endpoint address by your ingress IP address
-    echo "creating pool for deployment ..."
-    POOL_CMD="ketch pool add ${POOL} --ingress-service-endpoint ${INGRESS_ENDPOINT}" 
+    # Add a framework with ingress Traefik (default), replace ingress endpoint address by your ingress IP address
+    echo "creating framework for deployment ..."
+    POOL_CMD="ketch framework add ${POOL} --ingress-service-endpoint ${INGRESS_ENDPOINT}"
```
Comment (murphybytes) [inline ci/ketch-ci.sh:139]: s/POOL/FRAMEWORK/

File: ci/ketch-ci.sh [hunk 2]
```diff
@@ -147,7 +147,7 @@ if [ "$RESOURCE_CREATION" = true ] ; then
 
     # Create app
     echo "creating app for deployment ..."
-    APP_CMD="ketch app create ${APP_NAME} --pool ${POOL}" 
+    APP_CMD="ketch app create ${APP_NAME} --framework ${POOL}"
```
Comment (murphybytes) [inline ci/ketch-ci.sh:150]: `ketch app create` has been removed.

**Review by murphybytes — CHANGES_REQUESTED** [review]
There will be a lot of changes after you merge with main. I'll review after that

File: ci/ketch-ci.sh (hunk 1 above)
Comment (stinkyfingers) [reply ci/ketch-ci.sh]: I retained "pool" here. We should probably leave `pool` and `app create` here until we cut another release from the changes in this PR and other recent. I think - I'm not 100% sure that's the best way to handle it.

File: ci/ketch-ci.sh (hunk 2 above)
Comment (stinkyfingers) [reply ci/ketch-ci.sh]: Same as above.

**Review by murphybytes — APPROVED** [review]

File: ci/README.md [hunk 3]
```diff
@@ -9,22 +9,22 @@ The ingress controller (Traefik or Istio), cluster issuer, and cert-manager shou
 ### Usage
 
 ```
-Usage: ./ketch-ci.sh [-t --ketch-tag] [-o --pool] [-ig --ingress] [--endpoint] [-a --app] [-i --image] [-e --env] [-ig --ingress] [--registry-secret] [--ketch-yaml] [--procfile] [--skip-resource-creation]
+Usage: ./ketch-ci.sh [-t --ketch-tag] [-o --framework] [-ig --ingress] [--endpoint] [-a --app] [-i --image] [-e --env] [-ig --ingress] [--registry-secret] [--ketch-yaml] [--procfile] [--skip-resource-creation]
```
Comment (aleksej-paschenko) [inline ci/README.md:12]: shipa uses -k and --framework to specify a framework, I think we can follow the same convention
```
-k, --framework 
```

File: ci/README.md [hunk 4]
```diff
@@ -9,22 +9,22 @@ The ingress controller (Traefik or Istio), cluster issuer, and cert-manager shou
...
 ```
 
 | Flags | Descriptions |
 | ------ | ------ |
 |  -t, --ketch-tag  | Ketch version. Default is latest. |
-|  -o, --pool        |     Pool where your application should be deployed.|
+|  -o, --framework        |     Framework where your application should be deployed.|
```
Comment (aleksej-paschenko) [inline ci/README.md:18]: shipa uses -k and --framework to specify a framework, I think we can follow the same convention
```
-k, --framework (= "")
```

File: ci/README.md [hunk 5]
```diff
@@ -41,7 +41,7 @@ jobs:
     - stage: build
       script: ./build.sh     # build and push docker images
     - stage: deploy
-      script: ./ketch-ci.sh --ketch-tag v0.1.1 -a myapp -o mypool --endpoint 104.155.134.17 -i docker.io/shipasoftware/bulletinboard:1.0 --ingress traefik
+      script: ./ketch-ci.sh --ketch-tag v0.1.1 -a myapp -o myframework --endpoint 104.155.134.17 -i docker.io/shipasoftware/bulletinboard:1.0 --ingress traefik
```
Comment (aleksej-paschenko) [inline ci/README.md:44]: shipa uses -k and --framework to specify a framework, I think we can follow the same convention
```
-k, --framework (= "")
```

File: ci/README.md [hunk 6]
```diff
@@ -55,7 +55,7 @@ deployment:
   production:
     branch: "master"
     commands:
-      - ./ketch-ci.sh --ketch-tag v0.1.1 -a myapp -o mypool --endpoint 104.155.134.17 -i docker.io/shipasoftware/bulletinboard:1.0 --ingress traefik
+      - ./ketch-ci.sh --ketch-tag v0.1.1 -a myapp -o myframework --endpoint 104.155.134.17 -i docker.io/shipasoftware/bulletinboard:1.0 --ingress traefik
```
Comment (aleksej-paschenko) [inline ci/README.md:58]: `-k`

File: ci/README.md [hunk 7]
```diff
@@ -91,5 +91,5 @@ production:
     - branches
 
     script:
-      - ./ketch-ci.sh --ketch-tag v0.1.1 -a myapp -o mypool --endpoint 104.155.134.17 -i docker.io/shipasoftware/bulletinboard:1.0 --ingress traefik
-```
\ No newline at end of file
+      - ./ketch-ci.sh --ketch-tag v0.1.1 -a myapp -o myframework --endpoint 104.155.134.17 -i docker.io/shipasoftware/bulletinboard:1.0 --ingress traefik
```
Comment (aleksej-paschenko) [inline ci/README.md:94]: `-k`

File: ci/ketch-ci.sh [hunk 8]
```diff
@@ -136,7 +136,7 @@ if [ "$RESOURCE_CREATION" = true ] ; then
 
     # Add a pool with ingress Traefik (default), replace ingress endpoint address by your ingress IP address
     echo "creating pool for deployment ..."
-    POOL_CMD="ketch pool add ${POOL} --ingress-service-endpoint ${INGRESS_ENDPOINT}" 
+    POOL_CMD="ketch pool add ${POOL} --ingress-service-endpoint ${INGRESS_ENDPOINT}"
```
Comment (aleksej-paschenko) [inline ci/ketch-ci.sh:139]: ketch framework

File: ci/ketch-ci.sh [hunk 9]
```diff
@@ -147,7 +147,7 @@ if [ "$RESOURCE_CREATION" = true ] ; then
 
     # Create app
     echo "creating app for deployment ..."
-    APP_CMD="ketch app create ${APP_NAME} --pool ${POOL}" 
+    APP_CMD="ketch app create ${APP_NAME} --pool ${POOL}"
```
Comment (aleksej-paschenko) [inline ci/ketch-ci.sh:150]: framework

File: internal/deploy/parameters.go [hunk 10]
```diff
@@ -32,15 +32,15 @@ const (
...
+	FlagFramework      = "framework"
 	FlagRegistrySecret = "registry-secret"
 	FlagBuilder        = "builder"
 	FlagBuildPacks     = "build-packs"
 
 	FlagImageShort       = "i"
 	FlagDescriptionShort = "d"
 	FlagEnvironmentShort = "e"
-	FlagPoolShort        = "o"
+	FlagFrameworkShort   = "o"
```
Comment (aleksej-paschenko) [inline internal/deploy/parameters.go:43]: -k

**Review by aleksej-paschenko — CHANGES_REQUESTED** [review]
awesome! several small changes

File: ci/ketch-ci.sh (hunk 1 above)
Comment (stinkyfingers) [reply ci/ketch-ci.sh]: Nevermind me. Sounds like we'll create a release in short order which will legitimize `framework` in this script.

File: ci/README.md (hunk 3 above)
Comment (stinkyfingers) [reply ci/README.md]: Modified to `-k` in readme, ketch-ci script, and short flags (deploy/parameters.go).

File: ci/README.md (hunk 4 above)
Comment (stinkyfingers) [reply ci/README.md]: Modified to `-k` in readme, ketch-ci script, and short flags (deploy/parameters.go).

File: ci/README.md (hunk 5 above)
Comment (stinkyfingers) [reply ci/README.md]: Modified to `-k` in readme, ketch-ci script, and short flags (deploy/parameters.go).

**Review by aleksej-paschenko — APPROVED** [review]

**Pull request merged**